### ISSUE TITLE
rook case not working in ContiguityWeightsPolygons

### ISSUE THREAD

**Issue by dfolch** [pysal/pysal#815]
When running ```ps.weights._contW_binning.ContiguityWeightsPolygons``` for the the queen case all works fine, but for the rook case the following error comes up.

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
voronoi_utils.py in <module>()
    154 
    155     vor_polys = get_voronoi(pnts)
--> 156     w = shapely2w(vor_polys, 'rook')

voronoi_utils.py in shapely2w(polys, wttype)
    138         poly_index +=1
    139     pcollection = PolygonCollection(dict(zip(poly_ids, polys_pysal)))
--> 140     neighbors = ps.weights.Contiguity.ContiguityWeightsPolygons(pcollection, wttype=wttype).w
    141     return ps.W(neighbors)
    142 

pysal/pysal/weights/_contW_binning.py in __init__(self, collection, wttype)
    208         self.collection = collection
    209         self.wttype = wttype
--> 210         self.do_weights()
    211 
    212     def do_weights(self):

pysal/pysal/weights/_contW_binning.py in do_weights(self)
    293                 if polyId not in edgeCache:
    294                     iEdges = {}
--> 295                     iVerts = shpFileObject.get(polyId).vertices
    296                     nv = len(iVerts)
    297                     ne = nv - 1

NameError: global name 'shpFileObject' is not defined
```

When I swap out ```shpFileObject.get(xxxxxx).vertices``` for ```self.collection[xxxxxx].vertices``` on lines 295 and 323 the code runs to completion. Not sure if this might break something else though.

**Comment by ljwolf** [comment]
made the exact substitution you mention over in [my pdio feature branch](https://github.com/ljwolf/pysal/blob/pdio/pysal/weights/_contW_binning.py#L292) and haven't seen any issue with using it in the weights-from-iterable code (which looks like exactly what you're doing w/ shapely2w).

**Comment by ljwolf** [comment]
I think this is closed in `pysal/dev` by #821,  using the `from_iterable` contiguity classmethods.

```python
import pysal as ps
import shapely.geometry as geom

polygons = ps.open(ps.examples.get_path('south.shp')).read()
shapelys = [geom.asShape(g) for g in polygons]

ps.weights.Rook.from_iterable(shapelys)
<pysal.weights.Contiguity.Rook at 0x7f716afcfd50>
```
Is that satisfactory?

**Comment by dfolch** [comment]
Looks good.
